cmd/timescale-prometheus/main.go

```diff
@@ -406,19 +414,22 @@ func initElector(cfg *config) (*util.Elector, error) {
 	return &scheduledElector.Elector, nil
 }
 
-func migrate(cfg *pgclient.Config, appVersion pgmodel.VersionInfo) error {
-	shouldWrite, err := isWriter()
+func migrate(cfg *pgclient.Config, appVersion pgmodel.VersionInfo, leaseLock util.AdvisoryLock) error {
+	locked, err := leaseLock.GetAdvisoryLock()
```

Comment:
Don't we need both a leader lock and a schema lock here? Why is it safe to get rid of the leader lock check? That seems strange. I'd expect: first get a leader lock, then get a shared advisory lock, then iff a migration is necessary get a exclusive lock, then migrate.

Reply:
under the scheme where we grab an exclusive lock as the first thing we do when we try to migrate, it's gratuitously redundant: if we succeed in grabbing the migration lock we'll be the one to migrate regardless